data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/router/RouterFactory.java

```diff
@@ -22,6 +22,15 @@ public class RouterFactory {
 
     public Router createRouter(final Set<ConditionalRoute> routes) {
         final RouteEventEvaluator routeEventEvaluator = new RouteEventEvaluator(expressionEvaluator, routes);
-        return new Router(routeEventEvaluator, dataFlowComponentRouter, event -> event.getEventHandle().release(true));
+        return new Router(routeEventEvaluator, dataFlowComponentRouter,
+                event -> {
+                    // Release unrouted events ONLY if routes are defined.
+                    // If routes are not used in the config, then all
+                    // events will have zero size for the routes set and
+                    // we shouldn't be releasing such events here
+                    if (routes.size() > 0) {
```

Comment:
Thanks for adding the unit test. Let's also add one to verify that `release()` is not called when `routes.size() == 0`.

Reply:
This comment is not applicable  anymore because we have changed the code.

Reply:
I think we can drop this condition now. So this can be:

```
event -> event.getEventHandle().release(true)
```

Right?